python/cudf/cudf/core/frame.py

```diff
@@ -2688,7 +2688,13 @@ def acos(self):
                     1.5707963267948966,  1.266103672779499],
                     dtype='float64')
         """
-        return self._unaryop("acos")
+        if self.iloc[0].dtype == "int":
+            data = self.astype("float64")
+            data = data._unaryop("acos")
+        else:
+            data = self._unaryop("acos")
```

Comment:
```suggestion
            data = self.astype("float64")
        data = self._unaryop("acos")
```

Reply:
Will this change work? The data._unaryop("acos") calculates arcos for the integer values, and the self._unaryop("acos") is calculating this for non-integers.

Reply:
Ah, Sorry. updated the suggested change.

Reply:
Looks good, thank you! I'll commit the change :D